sale_order_net_price_control/models/sale_order_line.py

```diff
@@ -0,0 +1,71 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, _, fields
+from odoo.exceptions import ValidationError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    offer_price = fields.Boolean()
+    admits_offer_prices = fields.Boolean(related='product_id.admits_offer_prices', store=True)
+
+    def _compute_amount(self):
+        super()._compute_amount()
+        if not self.env.user.has_group('sale_order_net_price_control.group_sale_at_any_price'):
+
+            for record in self.filtered(lambda r: r.product_id and r.product_uom_qty > 0):
+
+                net_price_sale = record.price_subtotal / record.product_uom_qty
+
+                purchase_invoice = self.env['account.move.line'].search([
+                    ('product_id', '=', record.product_id.id),
+                    ('quantity', '>', 0),
+                    ('move_id.move_type', '=', 'in_invoice'),
+                    ('move_id.state', '=', 'posted'),
+                    ('display_type', '=', 'product'),
+                ], limit=1, order='name desc')
```

Comment:
This order is wrong, because we're looking for the "latest invoice", and `name` for an invoice line (`account.move.line`) is not invoice number (that it should've been wrong as well, as we can have several invoice sequences with different prefixes) but line description. Please use the proper condition instead (by date):
```suggestion
                ], limit=1, order='date desc')
```